src/common/core/core.js

```diff
@@ -287,6 +287,29 @@ define([
         }
     }
 
+    function ensureMongoCompatibleKey(input, nameOfInput, hiddenIsFine, isAsync) {
+        var error = null,
+            realInput = input;
+
+        if (hiddenIsFine === true && input[0] === '_') {
+            realInput = input.substr(1);
+        }
+
+        if (REGEXP.DOCUMENT_KEY.test(realInput) === false) {
+            error = new CoreIllegalArgumentError('Parameter ' + nameOfInput +
+                ' is not a valid key (cannot contain "." or "$"' +
+                (hiddenIsFine ? '' : 'and starts with "_"') + ').');
```

Comment:
This sentence is not grammatically correct if `hiddenIsFine == false`.

Reply:
True.